contracts/modules/cover/Cover.sol

```diff
@@ -793,22 +796,20 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     }
   }
 
-  function _isCoverAssetDeprecated(
-    uint32 deprecatedCoverAssetsBitmap,
-    uint8 assetId
-  ) internal pure returns (bool) {
-    return deprecatedCoverAssetsBitmap & (1 << assetId) > 0;
-  }
+  function isCoverAssetSupported(uint assetId, uint productCoverAssetsBitmap) internal view returns (bool) {
 
-  /// @dev Returns true if the assetsBitMap set is included in the coverAssetFallback set
-  /// @param assetsBitMap the assets bitmap for a product
-  /// @param coverAssetFallback  The coverAssetFallback as defined for the storage var with the same name
-  function areAssetsSupported(uint32 assetsBitMap, uint32 coverAssetFallback) public pure returns (bool) {
-    return assetsBitMap & bitwiseNegate(coverAssetFallback) == 0;
-  }
+    if (
+      // product does not use default cover assets
+      productCoverAssetsBitmap != 0 &&
+      // asset id is not in the product's cover assets bitmap
+      ((1 << assetId) & productCoverAssetsBitmap == 0)
+    ) {
+      return false;
+    }
+
+    Asset memory asset = pool().getAsset(assetId);
 
-  function bitwiseNegate(uint32 value) internal pure returns (uint32) {
-    return value ^ 0xffffffff;
+    return asset.isCoverAsset && !asset.isAbandoned;
```

Comment:
so coverASsetFallback is redundant now because you take as a fallback the assets from Pool that are cover assets.

Reply:
Yes, that was duplicate information anyway since it was present in a different form in the pool and has been removed now.